catalog/plugin/catalog-plugin-defaultsecurityattributevalues/src/main/java/ddf/catalog/metacard/security/DefaultSecurityAttributeValuesPlugin.java

```diff
@@ -0,0 +1,115 @@
+            HashMultimap.create();
+
+    ddf.security.Subject getSystemSubject() {
+        return org.codice.ddf.security.common.Security.getInstance()
+                .getSystemSubject();
+    }
+
+    /**
+     * Retrieves the system high attributes and the mapping of the system attribute names to
+     * metacard security markings. Returns a hash map of the metacard markings
+     * to the value of its corresponding system attribute.
+     *
+     * @return Map of the metacard security markings to the value of their corresponding system high
+     * attribute.
+     */
+    private Map<String, Attribute> getHighwaterSecurityMarkings() {
+        Map<String, Attribute> securityMarkings = new HashMap<>();
+        ddf.security.Subject system =
+                org.codice.ddf.security.common.Security.runAsAdmin(() -> getSystemSubject());
+        SecurityAssertion assertion = system.getPrincipals()
+                .oneByType(SecurityAssertion.class);
+        List<AttributeStatement> attributeStatements = assertion.getAttributeStatements();
+        for (AttributeStatement curStatement : attributeStatements) {
+            for (org.opensaml.saml.saml2.core.Attribute attribute : curStatement.getAttributes()) {
+                Collection<String> attributeNames =
+                        sysHighToMetacardAttributeMapping.get(attribute.getName());
+                if (!attributeNames.isEmpty()) {
+                    for (String attributeName : attributeNames) {
+                        HashSet<Serializable> values = attribute.getAttributeValues()
+                                .stream()
+                                .filter(curValue -> curValue instanceof XSString)
+                                .map(curValue -> ((XSString) curValue).getValue())
+                                .collect(Collectors.toCollection(HashSet::new));
+                        if (securityMarkings.containsKey(attributeName)) {
+                            values.addAll(securityMarkings.get(attributeName)
+                                    .getValues());
+                        }
+                        securityMarkings.put(attributeName,
+                                new AttributeImpl(attributeName,
+                                        (List<Serializable>) ImmutableList.<Serializable>copyOf(
+                                                values)));
+                    }
+                }
+            }
+        }
+        return securityMarkings;
+    }
+
+    @Override
+    public Metacard addDefaults(Metacard metacard) {
+        boolean hasSecurityAttr = sysHighToMetacardAttributeMapping.values()
+                .stream()
+                .anyMatch(attr -> metacard.getAttribute(attr) != null);
+
+        if (!hasSecurityAttr) {
+            Map<String, Attribute> securityMarkings = getHighwaterSecurityMarkings();
+            securityMarkings.values()
+                    .forEach(metacard::setAttribute);
```

Comment:
It wouldn't hurt to ensure that the mctype actually has the attribute descriptor, but in the case of security it _should_. Still, it's a good improvement to make.